Modules/xSharePoint/DSCResources/MSFT_xSPAntivirusSettings/MSFT_xSPAntivirusSettings.psm1

```diff
@@ -0,0 +1,113 @@
+        [parameter(Mandatory = $false)] [System.Boolean] $AllowDownloadInfected,
+        [parameter(Mandatory = $false)] [System.Boolean] $AttemptToClean,
+        [parameter(Mandatory = $false)] [System.UInt16] $TimeoutDuration,
+        [parameter(Mandatory = $false)] [System.UInt16] $NumberOfThreads,
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
+    )
+
+    Write-Verbose -Message "Getting antivirus configuration settings"
+
+    $result = Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+        $params = $args[0]
+        
+        [System.Reflection.Assembly]::LoadWithPartialName("Microsoft.SharePoint") | Out-Null
+
+        # Get a reference to the Administration WebService
+        $admService = [Microsoft.SharePoint.Administration.SPWebService]::ContentService
+
+        if ($null -eq $admService) { return $null }
+        
+        return @{
+            # Set the antivirus settings
+            AllowDownloadInfected = $admService.AntivirusSettings.AllowDownload
+            ScanOnDownload = $admService.AntivirusSettings.DownloadScanEnabled
+            ScanOnUpload = $admService.AntivirusSettings.UploadScanEnabled
+            AttemptToClean = $admService.AntivirusSettings.CleaningEnabled
+            NumberOfThreads = $admService.AntivirusSettings.NumberOfThreads
+            TimeoutDuration = $admService.AntivirusSettings.Timeout.TotalSeconds
+            InstallAccount = $params.InstallAccount
+        }
+    }
+
+    return $result
+}
+
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [parameter(Mandatory = $true)] [System.Boolean] $ScanOnDownload,
+        [parameter(Mandatory = $false)] [System.Boolean] $ScanOnUpload,
+        [parameter(Mandatory = $false)] [System.Boolean] $AllowDownloadInfected,
+        [parameter(Mandatory = $false)] [System.Boolean] $AttemptToClean,
+        [parameter(Mandatory = $false)] [System.UInt16] $TimeoutDuration,
+        [parameter(Mandatory = $false)] [System.UInt16] $NumberOfThreads,
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
+    )
+
+    Write-Verbose -Message "Setting antivirus configuration settings"
+
+    Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+        $params = $args[0]
+        
+        Write-Verbose -Message "Start update"
+        # Load the SharePoint Assembly, using old style for backward compatibility with V1
+        [System.Reflection.Assembly]::LoadWithPartialName("Microsoft.SharePoint") | Out-Null
+
+        # Get a reference to the Administration WebService
+        $admService = [Microsoft.SharePoint.Administration.SPWebService]::ContentService
+
+        # Set the antivirus settings
+        if ($params.ContainsKey("AllowDownloadInfected")) { 
+            Write-Verbose -Message "Setting Allow Download"
```

Comment:
Yep - we see the verbose output in the DSC extension in Azure for sure. I don't have an on-prem lab to test it with elsewhere but I've seen it using native commands like "Test-DSCConfiguration -Verbose" as well. Inside of the Invoke-xSharePointCommand statement we automatically bump the verbose output level up so we see everything. Makes for some large log files when you do something like set up a farm or a search topology for the first time, but we can drop custom messages out this way as well.